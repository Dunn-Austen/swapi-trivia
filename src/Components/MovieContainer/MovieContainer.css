* {
  margin: 0;
}

.movie-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0;
  width: 90%;
}

/* span {
  font-size: 50px;
  color:red;
} */
