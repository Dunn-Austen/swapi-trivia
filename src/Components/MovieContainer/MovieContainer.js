import React from 'react'
import './MovieContainer.css'
import MovieCard from '../MovieCard/MovieCard'

const MovieContainer = (props) => {
    const allMovies = props.movies.map(flick => {
        return (
            <MovieCard
                id={flick.episode_id}
                title={flick.title}
                number={flick.episode_id}
                key={flick.episode_id}
                releaseDate={flick.release_date}
                selectFlick={props.selectFlick}
            />
        )
    })
    return (
        <div className="movie-container">
            {allMovies}
        </div>
    )
}

export default MovieContainer


// Needs to include the UserData/Menu component (haven't decided on name / format yet), with logout and view favorites buttons
