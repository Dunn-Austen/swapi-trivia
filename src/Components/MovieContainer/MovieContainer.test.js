import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from '../MovieContainer/MovieContainer';

describe('MovieContainer', () => {   
    let wrapper;
    const mockSelect = jest.fn()
    beforeEach(() => {
        wrapper = shallow(<MovieContainer
            movies={[
                {
                    episode_id: 55,
                    title: 'Big Lebowski',
                    release_date: 1981
                },
                {
                    episode_id: 155,
                    title: 'Major League',
                    release_date: 2055
                }
            ]}
            user={{
                name: 'Foster',
                quote: 'Aim small miss small',
                ranking: 'Professional'
            }}
            select={mockSelect}
        />)
    })
  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});