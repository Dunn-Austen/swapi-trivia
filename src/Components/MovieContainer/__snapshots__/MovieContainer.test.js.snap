// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot with all data passed in correctly 1`] = `
<div
  className="movie-container"
>
  <MovieCard
    key="55"
    number={55}
    releaseDate={1981}
    title="Big Lebowski"
  />
  <MovieCard
    key="155"
    number={155}
    releaseDate={2055}
    title="Major League"
  />
</div>
`;
